bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -534,11 +484,26 @@ public void run() {
                         requestsCount = 1;
                     }
 
+                    journalStats.getForceWriteQueueSize().addCount(-requestsCount);
+
+                    // Sync and mark the journal up to the position of the last entry in the batch
+                    ForceWriteRequest lastRequest = localRequests.get(requestsCount - 1);
+                    syncJournal(lastRequest);
```

Comment:
Is it possible that there have two log files in the batch?

Reply:
+1
If the localRequests queue contains multiple journal files and we only sync the lastRequest's journal file, other journal files will skip sync.

Reply:
Actually, you're correct. We need to ensure all the files are closed before the response is triggered. Fixed it.

Reply:
Original logic will execute forceWrite before close, which will run bestEffortRemoveFromPageCache. If there have two different journal files in the batch, we only force write for the last file, do we need to do the force write for the another journal file? Does the `close` will do that?

Reply:
+1
It has two bugs:
- Those non-last journal files in the batch won't be removed from OS PageCache
- That non-last journal files channel in the batch will just call `close` instead of `fore(false)`, which can't ensure the data is flushed to the disk.

Reply:
I was actually 100% sure that close() implied an fsync, but that's not really the case.